doc/user/content/guides/node-js.md

```diff
@@ -0,0 +1,245 @@
+**S3** | [Write data from Node.js to S3](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/getting-started-nodejs.html) in an append-only fashion, use the Materialize [S3 Source](/sql/create-source/json-s3), to scan the S3 bucket for data and [listen for new data via SQS notifications](/sql/create-source/json-s3/#listening-to-sqs-notifications). If data is already sent to S3 and minute latency is not an issue, this is an economical and low-maintenance option.
+
+## Insert data into tables
+
+Most data in Materialize will stream in via a `SOURCE`, but a [`TABLE` in Materialize](https://materialize.com/docs/sql/create-table/) can be helpful for supplementary data. For example, use a table to join slower-moving reference or lookup data with a stream.
+
+**Basic Example:** [Insert a row](https://materialize.com/docs/sql/insert/) of data into a table named `countries` in Materialize.
+
+```js
+const { Client } = require('pg');
+const client = new Client('postgres://materialize@localhost:6875/materialize');
+
+const text = 'INSERT INTO countries(code, name) VALUES($1, $2);';
+const values = ['GH', 'GHANA'];
+
+async function main() {
+    await client.connect();
+    const res = await client.query(text, values);
+    console.log(res);
+}
+
+main();
+```
+
+## Manage sources, views, indexes
+
+Creation of sources, views and indexes in Materialize should be treated with the same care as schema creation and updates in a traditional database. These are typically done during Materialize deployment. However, a Node.js app is perfectly capable of executing these DDL statements.
+
+**Create a source from Node.js:**
+
+```js
+const { Client } = require('pg');
+const client = new Client('postgres://materialize@localhost:6875/materialize');
+
+async function main() {
+    await client.connect();
+    const res = await client.query(
+        `CREATE SOURCE market_orders_raw_2 FROM PUBNUB
+            SUBSCRIBE KEY 'sub-c-4377ab04-f100-11e3-bffd-02ee2ddab7fe'
+            CHANNEL 'pubnub-market-orders'`
+        );
+    console.log(res);
+}
+
+main();
+```
+
+See [`CREATE SOURCE`](/sql/create-source/) reference for more information.
+
+**Create a view from Node.js:**
+
+```js
+const { Client } = require('pg');
+const client = new Client('postgres://materialize@localhost:6875/materialize');
+
+async function main() {
+    await client.connect();
+    const res = await client.query(
+        `CREATE VIEW market_orders_2 AS
+            SELECT
+                val->>'symbol' AS symbol,
+                (val->'bid_price')::float AS bid_price
+            FROM (SELECT text::jsonb AS val FROM market_orders_raw)`
+        );
+    console.log(res);
+}
+
+main();
+```
+
+See [`CREATE VIEW`](/sql/create-view/) reference for more information.
```

Comment:
For more information, see [`CREATE VIEW`]().

(Consistent formatting makes it easier to search for terms when updating.)